url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+			// HiveLexer.g:322:11: ( 'august' ( 's' )? | 'aug' ( DOT )? )
+			int alt25=2;
+			int LA25_0 = input.LA(1);
+			if ( (LA25_0=='a') ) {
+				int LA25_1 = input.LA(2);
+				if ( (LA25_1=='u') ) {
+					int LA25_2 = input.LA(3);
+					if ( (LA25_2=='g') ) {
+						int LA25_3 = input.LA(4);
+						if ( (LA25_3=='u') ) {
+							alt25=1;
+						}
+
+						else {
+							alt25=2;
+						}
+
+					}
+
+					else {
+						int nvaeMark = input.mark();
+						try {
+							for (int nvaeConsume = 0; nvaeConsume < 3 - 1; nvaeConsume++) {
+								input.consume();
+							}
+							NoViableAltException nvae =
+								new NoViableAltException("", 25, 2, input);
+							throw nvae;
+						} finally {
+							input.rewind(nvaeMark);
+						}
+					}
+
+				}
+
+				else {
+					int nvaeMark = input.mark();
+					try {
+						input.consume();
+						NoViableAltException nvae =
+							new NoViableAltException("", 25, 1, input);
+						throw nvae;
+					} finally {
+						input.rewind(nvaeMark);
+					}
+				}
+
+			}
+
+			else {
+				NoViableAltException nvae =
+					new NoViableAltException("", 25, 0, input);
+				throw nvae;
+			}
+
+			switch (alt25) {
+				case 1 :
+					// HiveLexer.g:322:13: 'august' ( 's' )?
+					{
+					match("august"); 
+
+					// HiveLexer.g:322:25: ( 's' )?
+					int alt23=2;
+					int LA23_0 = input.LA(1);
+					if ( (LA23_0=='s') ) {
+						alt23=1;
+					}
+					switch (alt23) {
+						case 1 :
+							// HiveLexer.g:322:25: 's'
+							{
+							match('s'); 
+							}
+							break;
+
+					}
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:322:33: 'aug' ( DOT )?
+					{
+					match("aug"); 
+
+					// HiveLexer.g:322:39: ( DOT )?
+					int alt24=2;
+					int LA24_0 = input.LA(1);
+					if ( (LA24_0=='.') ) {
+						alt24=1;
+					}
+					switch (alt24) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+
+			}
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "AUGUST"
+
+	// $ANTLR start "SEPTEMBER"
+	public final void mSEPTEMBER() throws RecognitionException {
+		try {
+			int _type = SEPTEMBER;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:323:11: ( 'september' ( 's' )? | 'sep' ( DOT )? | 'sept' ( DOT )? )
+			int alt29=3;
+			int LA29_0 = input.LA(1);
+			if ( (LA29_0=='s') ) {
+				int LA29_1 = input.LA(2);
+				if ( (LA29_1=='e') ) {
+					int LA29_2 = input.LA(3);
+					if ( (LA29_2=='p') ) {
+						int LA29_3 = input.LA(4);
+						if ( (LA29_3=='t') ) {
+							int LA29_4 = input.LA(5);
+							if ( (LA29_4=='e') ) {
+								alt29=1;
+							}
+
+							else {
+								alt29=3;
+							}
+
+						}
+
+						else {
+							alt29=2;
+						}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 7144:7158, 8002:8016, 8328:8342.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.